### ISSUE TITLE
Failed to compile 14-iterative-search.Rmd

### ISSUE THREAD

**Issue by tanthiamhuat** [tidymodels/TMwR#147]
bookdown::render_book("index.Rmd", "bookdown::gitbook")

 Execution halted
Error in Rscript_render(f, render_args, render_meta, add1, add2) : 
  Failed to compile 14-iterative-search.Rmd

**Comment by juliasilge** [comment]
Hello @tanthiamhuat! 👋 Can you share any more details about the error that stopped the book from compiling for you? 

FWIW, this is a pretty complex project that take a while to compile ([about three hours on GH actions](https://github.com/tidymodels/TMwR/runs/2586791481)) and has a lot of dependencies. If you want to troubleshoot what you might be missing to get it to compile, you might take a look at the [DESCRIPTION](https://github.com/tidymodels/TMwR/blob/master/DESCRIPTION) file as well as the [GH action workflow](https://github.com/tidymodels/TMwR/blob/master/.github/workflows/bookdown.yaml). You might also consider running one chapter at a time that you are interested in, or chunk by chunk as you read along.

**Comment by tanthiamhuat** [comment]
it seems to be some issue on Chapter 14 which stops it.

**Comment by topepo** [comment]
The description file that is linked about shows that you need to install the development version of `finetune`. You can do that with the `remotes` package and 

```r
remotes::install_dev("finetune")
```

Give that a try. 

If there are still errors, I suggest running the Rmd file for that chapter incrementally so that you can give us some indication of where the problem occurs in chapter 14.

**Comment by tanthiamhuat** [comment]
same error:
Execution halted
Error in Rscript_render(f, render_args, render_meta, add1, add2) : 
  Failed to compile 14-iterative-search.Rmd

**Comment by topepo** [comment]
I suggest running the Rmd file for that chapter incrementally so that you can give us some indication of where the problem occurs in chapter 14.

**Comment by tanthiamhuat** [comment]
yes, I have tried running Rmd file for Chapter 14 before, and it gives the error below which I have no idea how to proceed from there:
![image](https://user-images.githubusercontent.com/9988941/119281930-382d9a00-bc6a-11eb-8a68-ec4cab3fddac.png)

**Comment by juliasilge** [comment]
By "incrementally", we mean not knitting the whole file at once, but rather running it chunk by chunk. 

For example, it looks like the final error happens at line 342. When you run it incrementally, chunk by chunk, are you able to create the visualization right before that, which should look like this?

![image](https://user-images.githubusercontent.com/12505835/119287362-983d2580-bc03-11eb-8f46-45e19b14d076.png)

**Comment by topepo** [comment]
Also, the error is related to _how_ we create the book. We use a nonstandard method of caching the results that checks to see if the same results were obtained from when we last cached. 

It might be that you used some random numbers in-between the chunks or perhaps you are on an older version of R (that used a different random number generator). 

In any case, that error is about the making of the book, rather than an error in the code or analysis.

**Comment by tanthiamhuat** [comment]
yes, I am able to get the above 2 plots before line 342, but after line 342, I got the below error:

![image](https://user-images.githubusercontent.com/9988941/119839079-78c13800-bf36-11eb-8727-91c3a1fb861f.png)

**Comment by tanthiamhuat** [comment]
R version 4.0.3 (2020-10-10) -- "Bunny-Wunnies Freak Out"
Copyright (C) 2020 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit)

**Comment by tanthiamhuat** [comment]
any update on it? I would love to be able to compile each chapter and get a complete book on Tidymodels. Thanks.

**Comment by juliasilge** [comment]
Can you tell us a little bit more about your use case for wanting to render the book locally? The error you are experiencing is from getting a different hash from the Bayesian results compared to what is checked in. You could change the hash that is hardcoded in your local copy, if needed.

**Comment by tanthiamhuat** [comment]
I would wish to have a complete understanding of Tidymodels from its very basic, as I find it important to grasp each of its concepts. Or if there is any complete book for us to download to read through it?

**Comment by juliasilge** [comment]
We are super happy about people reading the book and learning more about tidymodels from beginning to end! I am guessing you are aware that the entire book is available at <https://www.tmwr.org/>? Can you tell me a bit more about why you want to render the book locally? Are you rendering to HTML (like the site) or something other format?

**Comment by tanthiamhuat** [comment]
yes, I have thought of getting its pdf version from rendering the book, so that I can have its hardcopy and read it easily.
Thanks for the link, but does it contains a pdf which includes everything?

**Comment by juliasilge** [comment]
We don't expect the book to render to PDF right now TBH, especially given that there are video and GIF components. We plan to work on a paper version for publication once the book is complete, and generally have been recommending that folks "consume" the book via the rendered site at https://www.tmwr.org/ for now. If you prefer to render the entire book yourself but do get different results in this chapter in those lines, for now we recommend changing the hard coded hash.
